crates/block-producer/src/lib.rs

```diff
@@ -1,37 +1,27 @@
-use std::{sync::Arc, time::Duration};
-
-use batch_builder::batch::TransactionBatch;
-use miden_objects::transaction::ProvenTransaction;
-use tokio::sync::RwLock;
+use std::time::Duration;
 
 #[cfg(test)]
 pub mod test_utils;
 
 mod batch_builder;
 mod block_builder;
+mod domain;
 mod errors;
-mod state_view;
+mod mempool;
 mod store;
-mod txqueue;
 
 pub mod block;
 pub mod config;
 pub mod server;
 
-// TYPE ALIASES
-// =================================================================================================
-
-/// A vector that can be shared across threads
-pub(crate) type SharedRwVec<T> = Arc<RwLock<Vec<T>>>;
-
 // CONSTANTS
 // =================================================================================================
 
 /// The name of the block producer component
 pub const COMPONENT: &str = "miden-block-producer";
 
 /// The number of transactions per batch
-const SERVER_BATCH_SIZE: usize = 2;
+const SERVER_MAX_TXS_PER_BATCH: usize = 2;
 
 /// The frequency at which blocks are produced
 const SERVER_BLOCK_FREQUENCY: Duration = Duration::from_secs(10);
```

Comment:
Not for this PR, but let's reduce block times to 5 seconds.

Reply:
#564